expr/parse_filterql.go

```diff
@@ -0,0 +1,343 @@
+package expr
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	u "github.com/araddon/gou"
+	"github.com/araddon/qlbridge/lex"
+)
+
+// Parses Tokens and returns an request.
+func ParseFilterQL(filter string) (*FilterQL, error) {
+	l := lex.NewFilterQLLexer(filter)
+	m := FilterQLParser{l: l, FilterQLTokenPager: NewFilterQLTokenPager(l), buildVm: false}
+	return m.parse()
+}
+func ParseFilterQLVm(filter string) (*FilterQL, error) {
+	l := lex.NewFilterQLLexer(filter)
+	m := FilterQLParser{l: l, FilterQLTokenPager: NewFilterQLTokenPager(l), buildVm: true}
+	return m.parse()
+}
+
+type FilterQL struct {
+	Pos
+	Raw    string       // full original raw statement
+	Filter *Filters     // A top level filter
+	Limit  int          // Limit
+	Offset int          // Offset
+	Alias  string       // Non-Standard sql, alias/name of sql another way of expression Prepared Statement
+	With   u.JsonHelper // Non-Standard SQL for properties/config info, similar to Cassandra with, purse json
+}
+
+func NewFilterQL() *FilterQL {
+	req := &FilterQL{}
+	return req
+}
+
+type Filters struct {
```

Comment:
FilterPhrase? I dunno, `Filters` plural is a bit weird, but I see how it makes sense. Nice and succinct too. Probably just ignore me.